apps/passport-client/src/appHooks.ts

```diff
@@ -66,6 +66,14 @@ export function useDispatch(): Dispatcher {
   return dispatch;
 }
 
+export function useIsOffline(): boolean {
+  return useSelector<boolean>((s) => !!s.offline, []);
+}
+
+export function useStateContext(): StateContextValue {
+  return useContext(StateContext);
```

Comment:
If I'm understanding correctly, StateContext never changes during the lifecycle of the app (its members are objects or functions that get defined during bootstrapping). You can read from it using `getState()`, which returns the current state of the app, but unlike a `useState` hook this will not trigger a re-render if the underlying data changes. This is probably fine for the offline ticket data as we're not going to rely on re-rendering, but this pattern might not be safe for things that ought to re-render.